pkg/csi_driver/utils_test.go

```diff
@@ -669,3 +687,129 @@ func TestParseVolumeAttributes(t *testing.T) {
+			disallowedFlags: map[string]bool{
+				"debug_gcs": true,
+				"profile":   true,
+			},
+			expected: []string{},
+		},
+		{
+			name:            "no matching disallowed flags, results in noop",
+			mountOptions:    []string{"uid:1001", "gid=1002"},
+			disallowedFlags: map[string]bool{"debug_fuse": true},
+			expected:        []string{"uid:1001", "gid=1002"},
+		},
+	}
+
+	// Run test cases
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := removeDisallowedMountOptions(tc.mountOptions, tc.disallowedFlags)
+
+			if diff := cmp.Diff(actual, tc.expected); diff != "" {
+				t.Errorf("test %q failed: got %v, but want %v", tc.name, actual, tc.expected)
+			}
+		})
+	}
+}
+
+func TestGenerateDisallowedFlagsMap(t *testing.T) {
+	t.Parallel()
+	driver := initTestDriver(t, nil)
+	driver.config.FeatureOptions = &GCSDriverFeatureOptions{
+		FeatureGCSFuseProfiles: &FeatureGCSFuseProfiles{
+			EnableGcsfuseProfilesInternal: false,
+		},
+	}
+	cases := []struct {
+		name           string
+		image          string
+		expectedMap    map[string]bool
+		expectErr      error
+		enableProfiles bool
+	}{
+		{
+			name:           "nil sidecar image results in failed map generation",
+			image:          "",
+			expectedMap:    map[string]bool{},
+			expectErr:      fmt.Errorf("unable to get disallowed flags, sidecar image is empty"),
+			enableProfiles: false,
+		},
+		{
+			name:           "sidecar version predates storage profiles - results in disllowed profile flag",
+			image:          "v1.18.0-gke.1",
+			expectedMap:    map[string]bool{GCSFuseProfileFlag: true},
+			expectErr:      nil,
+			enableProfiles: true,
+		},
+		{
+			name:           "sidecar version supports storage profiles but feature is disabled - results in disllowed profile flag",
+			image:          "gke.gcr.io/gcs-fuse-csi-driver-sidecar-mounter:v1.19.3-gke.2@sha256:abcd",
+			expectedMap:    map[string]bool{GCSFuseProfileFlag: true},
+			enableProfiles: false,
+			expectErr:      nil,
+		},
+		{
+			name:           "sidecar version supports storage profiles and feature is enables - results in empty map",
+			image:          "gke.gcr.io/gcs-fuse-csi-driver-sidecar-mounter:v1.19.4-gke.2@sha256:abcd",
+			expectedMap:    map[string]bool{},
+			enableProfiles: true,
+			expectErr:      nil,
+		},
+	}
+
+	for _, test := range cases {
+		driver.config.FeatureOptions.FeatureGCSFuseProfiles.EnableGcsfuseProfilesInternal = false
+		if test.enableProfiles {
+			driver.config.FeatureOptions.FeatureGCSFuseProfiles.EnableGcsfuseProfilesInternal = true
+		}
+		resultMap, err := driver.generateDisallowedFlagsMap(test.image)
+		if err == nil && test.expectErr != nil {
+			t.Errorf("test %v failed:\ngot error nil, \nwanted error %q", test.name, test.expectErr)
+		}
+		if err != nil && (test.expectErr == nil || err.Error() != test.expectErr.Error()) {
+			t.Errorf("test %q failed:\ngot error, \nwanted error %q %q", test.name, test.expectErr, err)
+		}
```

Comment:
nit: this could be in helper function to be simpler, but up to you: 
```
func gotExpectedError(t *testing.T, wantErr error, gotErr error, testName string) {
	t.Helper()
	if wantErr == nil && gotErr == nil {
		return
	}
	if gotErr != nil && wantErr == nil {
		t.Errorf("%s: got error %v, want nil", testName, gotErr)
		return
	}
	if gotErr == nil && wantErr != nil {
		t.Errorf("%s: got nil, want error", testName)
		return
	}
	if gotErr.Error() != wantErr.Error() {
		t.Errorf("%s: got error message: %q, expected: %q", testName, gotErr.Error(), wantErr.Error())
	}
}
```